sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -139,6 +140,20 @@ class CSVSuite extends QueryTest with SharedSQLContext with SQLTestUtils {
     assert(result.schema === expectedSchema)
   }
 
+  test("test inferring booleans with custom values") {
+    val result = spark.read
+      .format("csv")
+      .option("header", "true")
```

Comment:
Sure, will make the change. If preferred, I can change the whole suite to the typesafe options as well.